shell/models/apps.deployment.js

```diff
@@ -14,7 +14,8 @@ export default class Deployment extends Workload {
   get replicaSetId() {
     const set = this.metadata?.relationships?.find((relationship) => {
       return relationship.rel === 'owner' &&
-            relationship.toType === WORKLOAD_TYPES.REPLICA_SET;
+            relationship.toType === WORKLOAD_TYPES.REPLICA_SET &&
+            relationship.message.split(':')[1].trim() > 0;
```

Comment:
This seems fragile, if the message does not contain `:` it'll throw an exception. It feels like we need to check all relationships and if one does contain the required text return the condition, if none contain the required text return the first one as before.

Reply:
I've updated the code to handle this case safely... Here is the modified one. 

```
export default class Deployment extends Workload {
  get replicaSetId() {
    const relationships = this.metadata?.relationships || [];
    
    // Find all relevant ReplicaSet relationships
    const replicaSetRelationships = relationships.filter((relationship) => 
      relationship.rel === 'owner' && relationship.toType === WORKLOAD_TYPES.REPLICA_SET
    );

    // Filter the ReplicaSets based on replicas > 0
    const activeReplicaSet = replicaSetRelationships.find((relationship) => {
      const replicasMatch = relationship.message?.match(/Replicas: (\d+)/);
      const replicas = replicasMatch ? parseInt(replicasMatch[1], 10) : 0;
      return replicas > 0;
    });

    // If no active ReplicaSet is found, fall back to the first one from the list
    const selectedReplicaSet = activeReplicaSet || replicaSetRelationships[0];

    return selectedReplicaSet?.toId?.replace(`${this.namespace}/`, '');
  }
```

Reply:
TBH this looks good. There's probably a way to do it in a single loop with `reduce` but there's not going to be a lot of looping or frequent calls.

I would suggest though moving the regex out to a `const` at the top of the file. Those can sometime initialise slowly so it's good practise reduce if possible.

const replicasRegEx = /.../
...
.match(replicasRegEx)

Reply:
Thank you for the suggestion! As recommended, I’ve moved the `replicasRegEx` constant to the top of the file.